KubeArmor/core/kubeUpdate.go

```diff
@@ -3011,6 +3017,23 @@ func (dm *KubeArmorDaemon) WatchConfigMap() cache.InformerSynced {
 				// update visibility for namespaces
 				dm.updateVisibilityWithCM(cm, updateEvent)
 
+				oldCM, okOld := oldObj.(*corev1.ConfigMap)
+				newCM, okNew := newObj.(*corev1.ConfigMap)
+				if !okOld || !okNew {
+					return
+				}
+
+				oldNs := oldCM.Data[cfg.ConfigUntrackedNs]
+				newNs := newCM.Data[cfg.ConfigUntrackedNs]
+
+				if oldNs != newNs {
+					dm.Logger.Printf("untracked namespaces updated: %s -> %s", oldNs, newNs)
+					dm.SystemMonitor.UpdateUntrackedNamespaces(newNs)
+
+					// re-apply visibility using updated untracked namespace list
+					dm.updateVisibilityWithCM(newCM, updateEvent)
+				}
```

Comment:
This would be a simpler implementation wdyt.
```suggestion
				cfg.Globalcfg.ConfigUntrackedNs  = strings.Split(cm.Data[cfg.ConfigUntrackedNs], ",")				
				dm.Logger.Printf("untracked namespaces : %s", cfg.Globalcfg.ConfigUntrackedNs )						
```

Reply:
That sounds reasonable
Using the current runtime state instead of comparing old/new ConfigMap objects does simplify this.

I was thinking of updating the UpdateFunc to rely on
`SystemMonitor.UntrackedNamespaces` as the old value and compare it with
the new ConfigMap value, something like this:

```go
newNs := cm.Data[cfg.ConfigUntrackedNs]
oldNs := strings.Join(dm.SystemMonitor.UntrackedNamespaces, ",")

if newNs != oldNs {
    dm.SystemMonitor.UpdateUntrackedNamespaces(newNs)
    dm.updateVisibilityWithCM(cm, updateEvent)
}

Reply:
Still, in this flow, we first convert oldNs from an array to a string and then convert newNs back into an array. This introduces a redundant conversion step, especially since the namespaces may be the same but in a different order. Moreover, we already handle visibility updates for every change with `dm.updateVisibilityWithCM`, so this extra conversion isn’t necessary.

Reply:
Got it — thanks for the detailed explanation

You're right: converting the runtime slice to a string and then parsing it
back introduces redundant conversions and can also lead to false updates
when the namespace order differs.

I’ll simplify the UpdateFunc to always forward the new ConfigMap value
directly to `UpdateUntrackedNamespaces` and move the normalization and
comparison logic inside `SystemMonitor`, keeping the watcher logic minimal.

Something along these lines:

```go
// UpdateFunc
newNs := cm.Data[cfg.ConfigUntrackedNs]
dm.SystemMonitor.UpdateUntrackedNamespaces(newNs)

// visibility updates are already handled via updateVisibilityWithCM

Reply:
please Review @Aryan-sharma11